modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/components/SubscriptionPlanRow.kt

```diff
@@ -0,0 +1,375 @@
+    isSelected: Boolean,
+    onClick: () -> Unit,
+    rowConfig: RowConfig,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .clip(RoundedCornerShape(8.dp))
+                .background(MaterialTheme.theme.colors.primaryUi01Active)
+                .padding(vertical = rowConfig.verticalPadding)
+                .then(
+                    if (isSelected) {
+                        Modifier.border(
+                            width = 2.dp,
+                            color = MaterialTheme.theme.colors.primaryField03Active,
+                            shape = RoundedCornerShape(8.dp),
+                        )
+                    } else {
+                        Modifier
+                    },
+                )
+                .clickable(
+                    enabled = !isSelected,
+                    onClick = onClick,
+                    role = Role.Button,
+                    indication = ripple(color = MaterialTheme.theme.colors.primaryIcon01),
+                    interactionSource = null,
+                )
+                .padding(16.dp),
+        ) {
+            CheckMark(
+                isSelected = isSelected,
+                modifier = Modifier
+                    .align(Alignment.CenterVertically)
+                    .size(22.dp),
+            )
+            Spacer(
+                modifier = Modifier.width(16.dp),
+            )
+            Column(
+                modifier = Modifier.weight(1f),
+                verticalArrangement = Arrangement.spacedBy(rowConfig.labelSpacing),
+            ) {
+                TextH30(
+                    text = plan.name,
+                )
+                TextP40(
+                    text = plan.price(),
+                    color = MaterialTheme.theme.colors.primaryText02,
+                    fontSize = 15.sp,
+                    lineHeight = 21.sp,
+                )
+            }
+            plan.pricePerPeriod(rowConfig)?.let {
+                TextP40(
+                    text = it,
+                    color = MaterialTheme.theme.colors.primaryText02,
+                    fontSize = 15.sp,
+                    lineHeight = 21.sp,
+                    modifier = Modifier.align(Alignment.CenterVertically),
+                )
+            }
+        }
+
+        val displayLabel = rowConfig.badgeText.text()
+        if (plan.tier == SubscriptionTier.Plus && plan.billingCycle == BillingCycle.Yearly && displayLabel.isNotEmpty()) {
+            Box(
+                modifier = Modifier
+                    .align(
+                        when (rowConfig.badgePosition) {
+                            BadgePosition.RIGHT -> Alignment.TopEnd
+                            BadgePosition.CENTER -> Alignment.TopCenter
+                        },
+                    )
+                    .offset(x = (-10).dp, y = (-10).dp)
```

Comment:
I've just noticed. `x` offset should be applied only in case of the `TopEnd` alignment.